### ISSUE TITLE
Device information "Unknown"

### ISSUE THREAD

**Issue by daodrg** [nielsbasjes/yauaa#191]
For the User Agent string:

`Mozilla/5.0 (Apple-iPhone7C2/1202.466; U; CPU like Mac OS X; en) AppleWebKit/420+ (KHTML, like Gecko) Version/3.0 Mobile/1A543 Safari/419.3`

there is no device information. Also `https://try.yauaa.basjes.nl/` could not retrive information about device and also OS information is not complete.

**Comment by nielsbasjes** [comment]
Thanks.
It seems to be one of these that is apparently not recognized yet.

http://www.enterpriseios.com/wiki/Complete_List_of_iOS_User_Agent_Strings

**Comment by nielsbasjes** [comment]
What do you think of this result?

```
- test:
    input:
      user_agent_string: 'Mozilla/5.0 (Apple-iPhone7C2/1202.466; U; CPU like Mac OS X; en) AppleWebKit/420+ (KHTML, like Gecko) Version/3.0 Mobile/1A543 Safari/419.3'
    expected:
      DeviceClass                          : 'Phone'
      DeviceName                           : 'Apple iPhone'
      DeviceBrand                          : 'Apple'
      DeviceVersion                        : 'iPhone 6'
      OperatingSystemClass                 : 'Mobile'
      OperatingSystemName                  : 'iOS'
      OperatingSystemVersion               : '??'
      OperatingSystemVersionMajor          : '??'
      OperatingSystemNameVersion           : 'iOS ??'
      OperatingSystemNameVersionMajor      : 'iOS ??'
      LayoutEngineClass                    : 'Browser'
      LayoutEngineName                     : 'AppleWebKit'
      LayoutEngineVersion                  : '420'
      LayoutEngineVersionMajor             : '420'
      LayoutEngineNameVersion              : 'AppleWebKit 420'
      LayoutEngineNameVersionMajor         : 'AppleWebKit 420'
      AgentClass                           : 'Browser'
      AgentName                            : 'Safari'
      AgentVersion                         : '3.0'
      AgentVersionMajor                    : '3'
      AgentNameVersion                     : 'Safari 3.0'
      AgentNameVersionMajor                : 'Safari 3'
      AgentLanguage                        : 'English'
      AgentLanguageCode                    : 'en'
      AgentSecurity                        : 'Strong security'
```

**Comment by daodrg** [comment]
I think this is much better. `DeviceClass` is essential for us. But it looks little a bit strange, that DeviceVersion is equal to `iPhone 6` as of the `Apple-iPhone7C2/1202.466` part of the ua string.

**Comment by nielsbasjes** [comment]
I've also updated https://try.yauaa.basjes.nl/ so you can more easily try it.

The iPhone7C2 (or iPhone7,2) is an internal code name for the iPhone 6.

These pages show the mappings of these codes that I use:
http://www.enterpriseios.com/wiki/Complete_List_of_iOS_User_Agent_Strings
https://www.theiphonewiki.com/wiki/Models
https://gist.github.com/adamawolf/3048717

**Comment by daodrg** [comment]
Sorry, I don't know about that mappings! I also tried updated online analyzer, it looks great. I tried the other strings from my list which had problems parsing, seems most trubles are gone away! What I can see, that OS version not always correctly shown. i.e. 


`Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:15.0) Gecko/20100101 Firefox/15.0.1`

OsVersion is unknown, but I think that it should be 15.0, or I mistaken?
